src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -401,14 +400,10 @@ func (r *ClusterReconciler) handlePodFinalizer(
 		if err != nil {
 			return fmt.Errorf("node-id annotation is not an integer: %w", err)
 		}
-		// get the broker list
-		redpandaPorts := networking.NewRedpandaPorts(rp)
-		clusterPorts := collectClusterPorts(redpandaPorts, rp)
-		clusterSvc := resources.NewClusterService(r.Client, rp, r.Scheme, clusterPorts, log)
 
-		pki, err := certmanager.NewPki(ctx, r.Client, rp, ar.getHeadlessServiceFQDN(), clusterSvc.ServiceFQDN(r.clusterDomain), r.Scheme, log)
+		pki, err := ar.getPKI()
 		if err != nil {
-			return fmt.Errorf("creating pki: %w", err)
+			return err
```

Comment:
why here we don't have context in error? Like in:

https://github.com/redpanda-data/redpanda/pull/11178/commits/21bc725e0d1c5ffd799e928962e5c2b9e35b0551#diff-208e6df471016ebc997b72c36741150b86b9023d2d8d698bbcb3a8e56f0f84d2R160

https://github.com/redpanda-data/redpanda/blob/21bc725e0d1c5ffd799e928962e5c2b9e35b0551/src/go/k8s/controllers/redpanda/cluster_controller.go#L160